pkgs/development/compilers/gcc-arm-embedded/7/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchurl, ncurses5, python27 }:
+
+stdenv.mkDerivation rec {
+  name = "gcc-arm-embedded-${version}";
+  version = "7-2018-q2-update";
+  subdir = "7-2018q2";
+
+  platformString =
+    if stdenv.isLinux then "linux"
+    else if stdenv.isDarwin then "mac"
+    else throw "unsupported platform";
+
+  urlString = "https://developer.arm.com/-/media/Files/downloads/gnu-rm/${subdir}/gcc-arm-none-eabi-${version}-${platformString}.tar.bz2";
+
+  src =
+    if stdenv.isLinux then fetchurl { url=urlString; sha256="0sgysp3hfpgrkcbfiwkp0a7ymqs02khfbrjabm52b5z61sgi05xv"; }
+    else if stdenv.isDarwin then fetchurl { url=urlString; sha256="0nc7m0mpa39qyhfyydxkkyqm7spfc27xf6ygi2vd2aym4r9azi61"; }
+    else throw "unsupported platform";
+
+  phases = [ "unpackPhase" "installPhase" "fixupPhase" ];
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r * $out
+  '';
+
+  dontPatchELF = true;
+  dontStrip = true;
+
+  preFixup = ''
+    find $out -type f | while read f; do
+      patchelf $f > /dev/null 2>&1 || continue
+      patchelf --set-interpreter $(cat ${stdenv.cc}/nix-support/dynamic-linker) "$f" || true
+      patchelf --set-rpath ${stdenv.lib.makeLibraryPath [ "$out" stdenv.cc.cc ncurses5 python27 ]} "$f" || true
```

Comment:
Please consider using [autoPatchelfHook](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/masterpdfeditor/default.nix#L14). This will make updating a bit easier and is more robust then the approach chosen here.

Reply:
Adding all libraries to buildInputs will enable the hook to add it to the RPATH.

Reply:
From your comments, I figured out I should do the following, but should I also modify/remove the `find` in `preFixup`?  

``` patch
--- a/pkgs/development/compilers/gcc-arm-embedded/7/default.nix
+++ b/pkgs/development/compilers/gcc-arm-embedded/7/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, ncurses5, python27 }:
+{ stdenv, fetchurl, ncurses5, python27, autoPatchelfHook }:
 
 stdenv.mkDerivation rec {
   name = "gcc-arm-embedded-${version}";
@@ -24,6 +24,10 @@ stdenv.mkDerivation rec {
     cp -r * $out
   '';
 
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  buildInputs = [ stdenv.cc.cc ncurses5 python27 ];
+
   dontPatchELF = true;
   dontStrip = true;
```

Reply:
In the best case `preFixup` should become unnecessary. You can also remove the explicit `phases` assignment. You only ever need to specify an installPhase unless there is an `make install`. `dontPatchELF` and `dontStrip` can also go.

Reply:
I am unable to get the build working using the changes you suggest, but I pushed them to a [separate `gcc-arm-embedded_autopatchelfhook` branch](https://github.com/prusnak/nixpkgs/tree/gcc-arm-embedded_autopatchelfhook), so I can return to that later. I don't want to dig into this any further, I just want to update the package and fix the `blackmagic` breakage for now.

Reply:
fair enough.

Reply:
cc @dezgeg it currently fails on 

```
$ patchelf --remove-rpath /nix/store/0bvwvmaz3rd8iz3zc3vyclwid0sfbbhg-gcc-arm-embedded-7-2018-q2-update/arm-none-eabi/lib/redboot-crt0.o
wrong ELF type
```

```
$ file /nix/store/0bvwvmaz3rd8iz3zc3vyclwid0sfbbhg-gcc-arm-embedded-7-2018-q2-update/arm-none-eabi/lib/redboot-crt0.o
/nix/store/0bvwvmaz3rd8iz3zc3vyclwid0sfbbhg-gcc-arm-embedded-7-2018-q2-update/arm-none-eabi/lib/redboot-crt0.o: ELF 32-bit LSB relocatable, ARM, EABI5 version 1 (SYSV), not stripped
```

https://dl.thalheim.io/e2GdCQGJnobRgFI7F3B73w/auto-patchelf-hook-error.log

I thought I saw some architecture related code in the hook. But I suppose this is not handled yet right?